src/eval.ml

```diff
@@ -165,6 +169,91 @@ and eq_modulo : term -> term -> bool = fun a b ->
   let res = try eq_modulo [(a,b)]; true with Exit -> false in
   if !log_enabled then log_eqmd (r_or_g res "%a == %a") pp a pp b; res
 
+(** [tree_walk t s] tries to match stack [s] against tree [t] *)
+and tree_walk : Dtree.t -> stack -> (term * stack) option = fun itree istk ->
+  (* Count maximum number of stored items to initialize variable array *)
+  let capacity = Dtree.iter ~do_leaf:(fun _ _ -> 0)
```

Comment:
Why not computing this only once when the tree is built, and record this in the tree structure?

Reply:
The capacity is now saved along with the tree in the symbol